doc/manual/manual.md

```diff
@@ -2476,3 +2476,249 @@ Currently, this function can only be used for LLVM verification, though
+name that SAW should use when printing `x`. `ptr_to_fresh_readonly` does the
+same, the variable cannot be written to.
+
+~~~~
+let ptr_to_fresh n ty = do {
+    x <- crucible_fresh_var n ty;
+    p <- alloc_init ty x;
+    return (x, p);
+};
+
+let ptr_to_fresh_readonly n ty = do {
+    x <- crucible_fresh_var n ty;
+    p <- alloc_init_readonly ty x;
+    return (x, p);
+};
+~~~~
+
+Finally, we define
+`oneptr_update_func : String -> LLVMType -> Term -> CrucibleSetup ()`.
+
+`oneptr_update_func n ty f` specifies the behavior of a function that takes
+a single pointer (with a printable name given by `n`) to memory containing a
+value of type `ty` and mutates the contents of that memory. The specification
+asserts that the contents of this memory after execution are equal to the value
+given by the application of `f` to the value in that memory before execution.
+
+~~~~
+let oneptr_update_func n ty f = do {
+    (x, p) <- ptr_to_fresh n ty;
+    crucible_execute_func [p];
+    crucible_points_to p (crucible_term {{ f x }});
+};
+~~~~
+
+#### The `quarterround` operation
+
+The C function we wish to verify has type
+`void s20_quarterround(uint32_t *y0, uint32_t *y1, uint32_t *y2, uint32_t *y3)`.
+
+The function's specification generates four symbolic variables and pointers to
+them in the precondition/setup stage. The pointers are passed to the function
+during symbolic execution via `crucible_execute_fun`. Finally, in the
+postcondition/return stage, the expected values are computed using the trusted
+Cryptol implementation and it is asserted that the pointers do in fact point to
+these expected values.
+
+~~~~
+let quarterround_setup : CrucibleSetup () = do {
+    (y0, p0) <- ptr_to_fresh "y0" (llvm_int 32);
+    (y1, p1) <- ptr_to_fresh "y1" (llvm_int 32);
+    (y2, p2) <- ptr_to_fresh "y2" (llvm_int 32);
+    (y3, p3) <- ptr_to_fresh "y3" (llvm_int 32);
+
+    crucible_execute_func [p0, p1, p2, p3];
+
+    let zs = {{ quarterround [y0,y1,y2,y3] }}; // from Salsa20.cry
+    crucible_points_to p0 (crucible_term {{ zs@0 }});
+    crucible_points_to p1 (crucible_term {{ zs@1 }});
+    crucible_points_to p2 (crucible_term {{ zs@2 }});
+    crucible_points_to p3 (crucible_term {{ zs@3 }});
+};
+~~~~
+
+#### Simple Updating Functions
+
+The following functions can all have their specifications given by the utility
+function `oneptr_update_func` implemented above, so there isn't much to say
+about them.
+
+~~~~
+let rowround_setup =
+    oneptr_update_func "y" (llvm_array 16 (llvm_int 32)) {{ rowround }};
+
+let columnround_setup =
+    oneptr_update_func "x" (llvm_array 16 (llvm_int 32)) {{ columnround }};
+
+let doubleround_setup =
+    oneptr_update_func "x" (llvm_array 16 (llvm_int 32)) {{ doubleround }};
+
+let salsa20_setup =
+    oneptr_update_func "seq" (llvm_array 64 (llvm_int 8)) {{ Salsa20 }};
+~~~~
+
+#### 32-Bit Key Expansion
+
+The next function of substantial behavior that we wish to verify has type
```

Comment:
Tiny wibble: maybe "type" -> "following prototype"? Since the C snippet is slightly more than a type.